src/WebJobs.Script/Description/CSharp/DirectSharedAssemblyProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public class DirectSharedAssemblyProvider : ISharedAssemblyProvider
+    {
+        private readonly Assembly _assembly;
+
+        public DirectSharedAssemblyProvider(Assembly assembly)
+        {
+            _assembly = assembly;
+        }
+
+        public bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            if (string.Compare(_assembly.GetName().Name, assemblyName, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Is `assemblyName` coming straight from the user? Maybe it's safer to create an AssemblyName from `assemblyName` and get the string back out to normalize it.